subiquity/common/filesystem/tests/test_gaps.py

```diff
@@ -82,6 +82,46 @@ def test_two_gaps(self):
         self.assertEqual(g2, gaps.at_offset(d, 60 << 20))
 
 
+class TestInteresects(unittest.TestCase):
+    def test_identity(self):
+        d = make_disk()
+        [gap] = gaps.parts_and_gaps(d)
+        self.assertEqual(gap, gaps.intersects(d, gap))
+
+    def test_front_used(self):
+        m, d = make_model_and_disk(size=100 << 20)
+        m.storage_version = 2
+        [gap] = gaps.parts_and_gaps(d)
+        make_partition(m, d, offset=0, size=20 << 20)
+        self.assertEqual(gaps.Gap(device=d, offset=20 << 20, size=79 << 20),
```

Comment:
I wonder if these tests would be clearer if the expected value was the gap returned by parts_and_gaps rather than constructed by hand (not sure tbh).

Reply:
I worry a bit about relying on in test the same methods used in the implementation, but it does look nice.
See new commits.